app/src/main/java/com/jerboa/ui/components/home/BottomNavActivity.kt

```diff
@@ -87,22 +88,29 @@ fun BottomNavActivity(
     val account = getCurrentAccount(accountViewModel)
     val ctx = LocalContext.current
     val scope = rememberCoroutineScope()
+    val homeViewModel: HomeViewModel = viewModel()
 
     val bottomNavController = rememberNavController()
     var selectedTab by rememberSaveable { mutableStateOf(NavTab.Home) }
-    val onSelectTab = { tab: NavTab ->
+    val onSelectTab = onSelectTab@{ tab: NavTab ->
         if (tab.needsLogin() && account == null) {
             loginFirstToast(ctx)
         } else {
             selectedTab = tab
+            val currentRoute = bottomNavController.currentDestination?.route
+            if (currentRoute == tab.name && currentRoute == NavTab.Home.name) {
```

Comment:
I think its more clear if you do `currentRoute == tab.name && tab == NavTab.Home`